### ISSUE TITLE
[valid-expect-in-promise] Error is reported even though promise is awaited

### ISSUE THREAD

**Issue by srmagura** [jest-community/eslint-plugin-jest#930]
**[Minimal repro](https://github.com/srmagura/eslint-plugin-jest-promise-repro)**

This code

```js
it("handles an error", async () => {
  const p = Promise.reject(new Error("test error")).then(undefined, (e) => {
    expect(e).toBeInstanceOf(Error);
    return "handled";
  });

  expect(await p).toBe("handled");
});
```

produces the error

```text
This promise should either be returned or awaited to ensure the expects in it's chain are called  jest/valid-expect-in-promise
```

The promise is being awaited so the lint rule should not error.

Versions:

- "eslint": "7.32.0"
- "eslint-plugin-jest": "25.0.1"
- Node 14.17.6

**Comment by G-Rath** [comment]
🤦 I knew I'd forgot a valid situation like that. Currently we're looking straight up for either a `return` or an `await` in the function body.
